CRM/Utils/PDF/Utils.php

```diff
@@ -258,8 +258,35 @@ private static function getDompdfOptions(): Options {
         $settings[$setting] = $value;
       }
     }
+
+    // core#4791 - Set cache dir to prevent files being generated in font dir
+    $cacheDir = self::getCacheDir($settings);
+    if ($cacheDir !== "") {
+      $settings['font_cache'] = $cacheDir;
+    }
     $options->set($settings);
     return $options;
   }
 
+  /**
+   * Get location of cache folder.
+   *
+   * @param array $settings
+   * @return string
+   */
+  private static function getCacheDir(array $settings): string {
+    // Use subfolder of custom font dir if it is writable
+    if (isset($settings['font_dir']) && is_writable($settings['font_dir'])) {
+      $cacheDir = $settings['font_dir'] . DIRECTORY_SEPARATOR . 'font_cache';
+    }
+    else {
+      $cacheDir = Civi::paths()->getPath('[civicrm.files]/upload/font_cache');
+    }
+    // Try to create dir if it doesn't exist or return empty string
+    if ((!is_dir($cacheDir)) && (!mkdir($cacheDir))) {
```

Comment:
I don't think it will make a difference in practice but mkdir should probably have the recursive parameter so if the parent folders don't exist it will make them. But `upload` will likely exist, and a custom folder will obviously exist. So let's ignore it.

I also tested that you don't lose your customizations if you've previously followed the instructions at https://docs.civicrm.org/installation/en/latest/general/unicode_pdf

Reply:
Yes, that probably would have been better, but as you say it probably won't make a difference in practice.

Thanks for testing this and merging.